### PR TITLE
redo cache switch

### PR DESCRIPTION
**Pull request by davidkelly**
When the ResourceManager was re-worked a bit, it broke the --cache switch (causing a segfault) when starting interface.  This re-implements it in a manner consistent with those changes.

### CODE REVIEW

**Comment by davidkelly** [comment]
Test Plan:

Same as https://github.com/highfidelity/hifi/pull/10803

**Comment by CainFoool** [comment]
Testing once built.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1375/HighFidelity-Beta-PR10897-936c21e8497b055feef74de6cbcd8620eda1c91c.exe

**Comment by CainFoool** [comment]
# Test Passed
Test | Result
-------- | --------
install the PR. | Pass
it will launch (by default), that's fine. Shut it down | Pass
create a directory (initially empty) to use as the cache directory | Pass
launch interface with --cache <full path to that directory> | Pass
after launch, you will see the data8, prepared, and ktx_cache entries in the directory, and that they were written after launch | Pass
in AppData/Local/High Fidelity - PR10803/Interface, you will see those same 3 directories, but the times will all be from the earlier launch | Pass (10897)

Working as expected, passed.

Let me know if this needs further testing.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Comment by sethalves** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1377/HighFidelity-Beta-PR10897-936c21e8497b055feef74de6cbcd8620eda1c91c.exe

**Comment by davidkelly** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1380/HighFidelity-Beta-PR10897-936c21e8497b055feef74de6cbcd8620eda1c91c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1380/HighFidelity-Beta-PR10897-936c21e8497b055feef74de6cbcd8620eda1c91c.dmg

File: libraries/shared/src/PathUtils.cpp [hunk 1]
```diff
@@ -34,12 +35,8 @@ QString PathUtils::getAppDataPath() {
...
 }
 
-QString PathUtils::getAppLocalDataPath(const QString& overridePath /* = "" */) {
-    static QString overriddenPath = "";
-    // set the overridden path if one was passed in
-    if (!overridePath.isEmpty()) {
-        overriddenPath = overridePath;
-    }
+QString PathUtils::getAppLocalDataPath() {
+    QString overriddenPath = qApp->property(hifi::properties::APP_LOCAL_DATA_PATH).toString();
```
Comment (zfox23) [inline libraries/shared/src/PathUtils.cpp]: Isn't this initialized as a non-empty string (and thus the path would always be non-empty), or am I reading this incorrectly?

Comment (zfox23) [inline libraries/shared/src/PathUtils.cpp]: `const char* APP_LOCAL_DATA_PATH = "com.highfidelity.appLocalDataPath";`

Comment (davidkelly) [inline libraries/shared/src/PathUtils.cpp]: no, it is empty to begin with

**Pull request merged**